web/src/utils/port.ts

```diff
@@ -1,29 +1,56 @@
-import { getService } from 'port-numbers';
+import * as portnumbers from 'port-numbers';
+import { Config } from '../components/config';
 
-export const getProtectedService = (p: number) => {
+export const getService = (p: number, config: Config) => {
+  if (!config.PortNaming.Enable) {
+    return null;
+  }
+  if (!!config.PortNaming.PortNames[p.toString()]) {
+    return config.PortNaming.PortNames[p.toString()];
+  }
```

Comment:
Using a map will makes things easier here:

```suggestion
  if (config.PortNaming.PortNames.has(p)) {
    return config.PortNaming.PortNames.get(p);
  }
```